Is this a bug in the database/sql layer? Panic on Exec of empty string

Passing an empty to a string causes a panic in the database/sql layer. Is there some initialization I'm missing? I see nothing in the docs that state this should panic.

```
~/repos/go-empty-sql-statement/src/github/com/otoolep/go-empty-sql-statement (master)$ cat main.go         
```
```go                                          
package main                                                                                                                                         
                                                                                                                                                     
import (                                                                                                                                             
        "database/sql"                                                                                                                               
        "fmt"                                                                                                                                        
        "log"                                                                                                                                        
        "os"                                                                                                                                         
                                                                                                                                                     
        _ "github.com/mattn/go-sqlite3"                                                                                                              
)                                                                                                                                                    
                                                                                                                                                     
func main() {                                                                                                                                        
        os.Remove("foo.bar")                                                                                                                         
                                                                                                                                                     
        db, err := sql.Open("sqlite3", "foo.bar")                                                                                                    
        if err != nil {                                                                                                                              
                log.Fatalf("failed to open database: %s", err.Error())                                                                               
        }                                                                                                                                            
                                                                                                                                                     
        results, err := db.Exec("")
        if err != nil {
                log.Fatalf("failed to get results: %s", err.Error())
        }
        li, err := results.LastInsertId()
        fmt.Println(li, err)
}
```
```
~/repos/go-empty-sql-statement/src/github/com/otoolep/go-empty-sql-statement (master)$ go run main.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4fa135]

goroutine 1 [running]:
database/sql.driverResult.LastInsertId(0x6513a0, 0xc000144000, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/philip/.gvm/gos/go1.15/src/database/sql/sql.go:3266 +0x75
main.main()
        /home/philip/repos/go-empty-sql-statement/src/github/com/otoolep/go-empty-sql-statement/main.go:24 +0xfd
exit status 2
~/repos/go-empty-sql-statement/src/github/com/otoolep/go-empty-sql-statement (master)$ 
```

--- a/sqlite3.go
+++ b/sqlite3.go
@@ -828,6 +828,10 @@ func (c *SQLiteConn) exec(ctx context.Context, query string, args []namedValue) 
 		tail := s.(*SQLiteStmt).t
 		s.Close()
 		if tail == "" {
+			if res == nil {
+				// https://github.com/mattn/go-sqlite3/issues/963
+				res = &SQLiteResult{0, 0}
+			}
 			return res, nil
 		}
 		query = tail

--- a/sqlite3_test.go
+++ b/sqlite3_test.go
@@ -1943,6 +1943,7 @@ var tests = []testing.InternalTest{
 	{Name: "TestManyQueryRow", F: testManyQueryRow},
 	{Name: "TestTxQuery", F: testTxQuery},
 	{Name: "TestPreparedStmt", F: testPreparedStmt},
+	{Name: "TestExecEmptyQuery", F: testExecEmptyQuery},
 }
 
 var benchmarks = []testing.InternalBenchmark{
@@ -2271,6 +2272,25 @@ func testPreparedStmt(t *testing.T) {
 		}()
 	}
 	wg.Wait()
+}
+
+// testEmptyQuery is test for validating the API in case of empty query
+func testExecEmptyQuery(t *testing.T) {
+	db.tearDown()
+	res, err := db.Exec(" -- this is just a comment ")
+	if err != nil {
+		t.Fatalf("empty query err: %v", err)
+	}
+
+	_, err = res.LastInsertId()
+	if err != nil {
+		t.Fatalf("LastInsertId returned an error: %v", err)
+	}
+
+	_, err = res.RowsAffected()
+	if err != nil {
+		t.Fatalf("RowsAffected returned an error: %v", err)
+	}
 }
 
 // Benchmarks need to use panic() since b.Error errors are lost when

